api/v1beta1/remotesecret_types.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RemoteSecretSpec defines the desired state of RemoteSecret
+type RemoteSecretSpec struct {
+	Secret LinkableSecretSpec `json:"secret"`
+	Target RemoteSecretTarget `json:"target"`
+}
+
+type RemoteSecretTarget struct {
```

Comment:
What is Environment?
A name of the secret with kubeconfig for connection to other cluster?

Reply:
Environment is the appstudio environment. this is being talked about as our next step in developing this functionality.

Reply:
I see.
I would not like we have direct dependencies on appstudio logical entities here. It is all about physical location (namespace + (maybe) cluster), let external entities care about this mapping, not RemoteSecret.

Reply:
yes, agreed. I removed the hints of the environment from the code.